### ISSUE TITLE
#comments method on compiled method only considered the method comment

### ISSUE THREAD

**Issue by aranega** [guillep/pharo-core#3068]
When you ask a compiled method for its comments, the `#comments` method only returns the first comment directly set under the method signature (the method comment). Other comments are not considered by the method. For example, on the following method:

```smalltalk
DummyClass>>#methodToRefactorWithGuardClause
	"Returned comment"
	1 + 1.
	"ignored comment"
	1 + 1.
```

This call only returns the first comment.

```smalltalk
"only the first comment is returned"
(DummyClass>>#methodToRefactorWithGuardClause) comments
```

**Comment by bencoman** [comment]
Interesting idea.  Though it is #comment singular not #comments plural so the behaviour does make some sense.  What is your use case?  

btw, I put a ``self haltOnce`` in the method and can't seem to get that triggered. Any idea where that method is invoked from?

**Comment by aranega** [comment]
Actually there is a `#comments` selector on `CompiledMethod`, the code is the following:

```smalltalk
comments
	"Answer a string representing the comments in the method. Return an empty collection if the method's source code does not contain a comment."

	^ self ast comments collect: [ :comment | comment contents ]
```

It is said that it returns all the comments from the method, so it sounds that all the comments contained in the method should be returned.

**EDIT>** Sorry, I forgot about the use case. We have an intern here that is working on tests and make use of all the comments defined in the method (as far as I understood). But also when you ask for `comments` on a method, it would sound lo logical to have of them (I mean, it kinda sounds more logical)

**Comment by MarcusDenker** [comment]
see #2950. Maybe we should go back to the handwritten text parser for now

**Comment by bencoman** [comment]
@aranega, okay, looking at that method definition, what you says makes sense.  So this would give you what you want...
```
comments
	^ self ast allChildren flatCollect: [:el| el comments contents]
```
The next question then is what would be the impact on existing users #comments ?

**Comment by kasperosterbye** [comment]
I am still in the process of trying to make the pull request auto close an issue. No luck so far. This issue is solved using PR #4502

**Comment by MarcusDenker** [comment]
the PR needs to have "fixes #3068" as the description (as part of the PR commit message). 

I forget it often, too. Maybe we should pre-fill the comment with that if the commit is done with the tool for Issue tracker related commits

**Comment by Ducasse** [comment]
I think that iona did it.
